wicket-native-websocket/wicket-native-websocket-javax/src/main/java/org/apache/wicket/protocol/ws/javax/JavaxUpgradeHttpRequest.java

```diff
@@ -75,14 +81,21 @@ public JavaxUpgradeHttpRequest(final Session session, EndpointConfig endpointCon
 			userProperties = endpointConfig.getUserProperties();
 		}
 
-		this.httpSession = (HttpSession) userProperties.get("session");
-		this.headers = (Map<String, List<String>>) userProperties.get("headers");
+		this.httpSession = (HttpSession) userProperties.get(SESSION);
+		this.headers = (Map<String, List<String>>) userProperties.get(HEADERS);
 		this.queryString = session.getQueryString();
 		this.userPrincipal = session.getUserPrincipal();
 		Object requestURI = session.getRequestURI();
 		this.requestUri = requestURI != null ? requestURI.toString() : "";
-		this.contextPath = httpSession.getServletContext().getContextPath();
-
+		if (httpSession == null)
+		{
+			this.contextPath = "/";
```

Comment:
@martin-g I'm not sure this is correct or if this path is used anywhere else.. I was looking into code base and didn't see any other object from where we could take the path. We are getting this NPE from time to time on our test severs...

Reply:
I am not sure either. It should be OK.